sisl/sparse.py

```diff
@@ -1352,6 +1352,37 @@ def copy(self, dims=None, dtype=None):
 
         return new
 
+    def convert(self, scale, dtype=None):
+        """ Convert sparse matrix data to a different dimension by mixing with specified coefficients
+
```

Comment:
We should add an example her, probably something similar to the `test_convert` test would suffice to clarify its use and what it does.

Reply:
I agree. Was also thinking the variable could be called `coefficients` instead of `scale`. What do you think is more descriptive?

Reply:
or `transformation_matrix`?

Reply:
Maybe this low-level function for `SparseCSR` should even be changed to `transform`? We can still use `convert` for the high-level spin changes.

Reply:
I think you are correct. `transform` is the better word! Perhaps that would allow the argument to be named `matrix` ? `.transform(matrix=...)` wouldn't that suffice in telling that matrix is the transformation matrix?

Then it could be implemented like this:

```python
matrix = np.asarray(matrix)
new._D = old._D.dot(matrix.T)
```
which means that the matrix is LHS operated.

Reply:
Good, let's go for this.